c10/util/typeid.cpp

```diff
@@ -20,36 +20,62 @@ C10_EXPORT void _ThrowRuntimeTypeLogicError(const string& msg) {
 
 } // namespace detail
 
+//
+// Specializations of TypeMetaData* TypeMeta::_typeMetaDataInstance<T>() -
+// these return singletons, effectively interning TypeMetaData instances.
+// TypeMeta objects are just handles to these interned pointers.
+//
+// First the instances corresponding to ScalarType types, then others
+// needed by the core - we use the user macro CAFFE_KNOWN_TYPE(T) for
+// the latter
+//
+
+//
+// TypeMeta/ScalarType bridge
+//
+
+// Canonical TypeMetaData instances for C++ types enumerated by ScalarType
+namespace detail {
+static TypeMetaData scalar_type_metas[] = {
```

Comment:
Top level static. Are you sure you aren't subject to SIOF here?

Reply:
Yeah, good point. Final version of this PR simplifies this array out of existence, and #44965 shouldn't have any top level statics.